cmd/ci-operator/main.go

```diff
@@ -824,6 +833,36 @@ func (o *options) writeMetadataJSON() error {
 		return nil
 	}
 
+	// Try to find the custom prow metadata file. We assume that there's only one. If there's more than one,
+	// we'll just use the first one that we find.
+	var customProwMetadataFile string
+	filepath.Walk(o.artifactDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
This metadata file will be generated from the job, right? Can we find another way of detecting it? Maybe we can allocate a specific env var for the custom metadata file, instead of trying to find it. Something like `$CUSTOM_PROW_METADATA_FILE`.

@stevekuznetsov @petr-muller Any thoughts on that?

Reply:
One thing that makes it hard is the indirection layer in the template. `ci-operator` does not know the name of the directory where the template will put its artifacts.

Another thing is that ci-operator can run multiple targets, and each target can (in theory) create different artifacts, so having a single location would be tricky. So I think we'll eventually want to handle multiple of these anyways, which seems fine to me to do incrementally and half-ass it with the "first found, first used" approach here.